lib/util/clone.ts

```diff
@@ -1,11 +1,11 @@
-import { quickStringify } from './stringify';
-
 /**
  * Creates a deep clone of an object.
- * @deprecated Use {@link structuredClone} instead.
  * @param input The object to clone.
  */
-export function clone<T>(input: T | null = null): T {
-  const stringifiedInput = quickStringify(input);
-  return stringifiedInput ? JSON.parse(stringifiedInput) : null;
+export function clone<T = unknown>(input: T): T {
+  const res = structuredClone(input);
+  if (typeof input === 'object' && input !== null) {
+    Object.setPrototypeOf(res, Object.getPrototypeOf(input));
+  }
```

Comment:
Revert this part for now?

Reply:
No, it will break http layer

Reply:
So how can we simply _wrap_ structuredClone now?

Reply:
We have to explicitly set prototypes for cases when it will fail otherwise, i.e. gitea and one more place

Reply:
Tests fail, you mean? Not actual functionality failing?

Reply:
Actual functionality will fail, since `is.plainObject` doesn't work on objects cloned by `structuredClone`. Hence this defensive prototype setting.

Reply:
Should we just be reverting the structuredClone change and using our util/clone with stringifying?

Reply:
Yes, I would like to switch implementation right in this PR, I don't think it's a big problem.
But if you want, I could do this in separate PR.

Reply:
@JamieMagee do you object to us removing structuredClone and replacing with stringify/parsing?

Reply:
yes, i prefer to stick on JSON clone, because of the current issues with structured clone. setting prototype manually removes the structured clone performance improvement. so it has no benefits.

Reply:
So replace every `structuredClone()` with `clone()` from `util/clone`?

Reply:
My intention to remove `JSON.parse` is because it doesn't work well with #20553